addons/website_slides/controllers/main.py

```diff
@@ -489,32 +489,35 @@ def submit_quiz(self, slide_id, answer_ids,**kw):
         good_answers = request.env['slide.answer'].search([('id', 'in', answer_ids), ('is_correct', '=', True)])
         bad_answers = request.env['slide.answer'].browse(answer_ids) - good_answers
         slide_partner = request.env['slide.slide.partner'].search([('slide_id', '=', slide_id), ('partner_id', '=', request.env.user.partner_id.id)])
-        possible_points = [slide.quiz_first_attempt_reward, slide.quiz_second_attempt_reward, slide.quiz_third_attempt_reward, slide.quiz_fourth_attempt_reward]
         points = 0
         if not slide_partner:
-            slide_partner = request.env['slide.slide.partner'].sudo().create({
-                "slide_id": slide.id,
-                "partner_id": request.env.user.partner_id.id,
-                "channel_id": slide.channel_id.id,
-                "completed": False
-            })
+            slide.action_set_viewed()
         if not slide_partner.completed:
-            if slide_partner.quiz_attempts_count < len(possible_points):
-                points = possible_points[slide_partner.quiz_attempts_count]
-            else:
-                points = possible_points[-1]
+            points = self._get_quiz_points(slide, slide_partner.quiz_attempts_count)
             slide_partner.sudo().write({
                 'quiz_attempts_count': slide_partner.quiz_attempts_count if not bad_answers else slide_partner.quiz_attempts_count + 1,
                 'points_won': points if not bad_answers else 0,
                 'completed': not bad_answers
             })
+            user = {}
```

Comment:
I would name it user_karma or karma_progress_bar as it's information about user karma for the progression bar and not about the user itself.